### PR TITLE
[measure tool] Fix mix up in area/distance units when reactivating the measure tool after changing the unit type

### LINKED ISSUE #43732: Measure Line wrong unit for displayed heading on first use
QGIS 3.18 on Debian GNU/Linux Bullseye pre-release

When using the measure tool the "Segments [feet]" text header sometimes appears when measure tool default "map units" option is active, but the displayed numbers are actually meters. Selecting any unit option from the measure tool dialog and back to "map units" the displayed number is in feet matching the text header. It happens occasionally, not every time.

1. start QGIS
2. toggle editing (on)  - I do this to get the snapping from a point on a Point layer.
3. Measure Line
4. click near point to snap, click again some other place.

Problem: Measure Line heading shows "Segments [feet]" however numbers displayed are in meters by value. Project default is set to feet. See screenshots
QGIS318MeasureLineBugUnits
.

Work-around: Always select away from "map units" on Measure Line to some other unit and then back to "map units". Units and headings are then displayed correctly.

This does not always happen, but I am able to reproduce by closing QGIS, open the project, and use the Measure Line tool. Some other situations also make it have this weird behavior, but closing and opening the project is a way to reproduce the issue.

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR fixes https://github.com/qgis/QGIS/issues/43732 , whereas the measure tool dialog would not properly reset/refresh settings when reactivating the tool leading to mixup in area/distance unit type.

### CODE REVIEW

**Comment by agiudiceandrea** [comment]
Has also https://github.com/qgis/QGIS/issues/26723 been fixed by this PR?

File: src/app/qgsmeasuredialog.cpp [hunk 1]
```diff
@@ -161,6 +161,31 @@ void QgsMeasureDialog::updateSettings()
   mAreaUnits = QgsProject::instance()->areaUnits();
   mDa.setSourceCrs( mCanvas->mapSettings().destinationCrs(), QgsProject::instance()->transformContext() );
 
+  if ( mMeasureArea )
```
Comment (nyalldawson) [inline src/app/qgsmeasuredialog.cpp:164]: Can this duplicate logic be merged with that in QgsMeasureDialog::unitsChanged ? Ie extract this part out to a separate function and call that function both here and in QgsMeasureDialog::unitsChanged.

**Comment by nirvn** [comment]
@agiudiceandrea , no, that issue is a distinct one, not addressed by this PR.

File: src/app/qgsmeasuredialog.cpp (hunk 1 above)
Comment (nirvn) [reply src/app/qgsmeasuredialog.cpp:164]: @nyalldawson , done, thanks for the review.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**